ydb/core/persqueue/pq_impl.cpp

```diff
@@ -1176,22 +1283,22 @@ void TPersQueue::AggregateAndSendLabeledCountersFor(const TString& group, const
 
 void TPersQueue::Handle(TEvPQ::TEvPartitionLabeledCounters::TPtr& ev, const TActorContext& ctx)
 {
-    auto it = Partitions.find(ev->Get()->Partition);
-    Y_ABORT_UNLESS(it != Partitions.end());
+    const auto partitionId = ev->Get()->Partition;
+    auto& partition = GetPartitionInfo(partitionId);
     const TString& group = ev->Get()->LabeledCounters.GetGroup();
-    it->second.LabeledCounters[group] = ev->Get()->LabeledCounters;
+    partition.LabeledCounters[group] = ev->Get()->LabeledCounters;
     Y_UNUSED(ctx);
 //  if uncommented, all changes will be reported immediatly
 //    AggregateAndSendLabeledCountersFor(group, ctx);
 }
 
 void TPersQueue::Handle(TEvPQ::TEvPartitionLabeledCountersDrop::TPtr& ev, const TActorContext& ctx)
 {
-    auto it = Partitions.find(ev->Get()->Partition);
-    Y_ABORT_UNLESS(it != Partitions.end());
+    const auto partitionId = ev->Get()->Partition;
+    auto& partition = GetPartitionInfo(partitionId);
     const TString& group = ev->Get()->Group;
```

Comment:
The same situation. Do not process labeled counters from shadow partitions.

Reply:
done
https://github.com/ydb-platform/ydb/pull/1037/commits/5090d78400d2f4c82c3f358033a1b8b725aea96b